lib/puppet/configurer.rb

```diff
@@ -472,6 +480,16 @@ def find_functional_server
   end
   private :find_functional_server
 
+  def last_used_environment
+    last_run_environment = Puppet::Util::Yaml.safe_load_file(Puppet[:lastrunfile])['version']['environment']
```

Comment:
Wondering about cases where the file doesn't exist (such as a new install) or the file exists but doesn't have the `environment` entry (such as if the file was written by an older puppet version). In the first case, I think it's fine to fallback to `Puppet[:environment]`. In the second case, maybe we fallback to loading the `lastrunreport` (if it exists) and grabbing the environment from there?